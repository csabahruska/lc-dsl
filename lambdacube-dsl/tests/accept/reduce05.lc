data AB = A | B

render fb choice = let
  rasterCtx         = TriangleCtx CullNone PolygonFill NoOffset LastVertex
  mask              = case choice of
                        A -> V4B False True False False
                        B -> V4B True False False False
  fragmentCtx       = AccumulationContext (ColorOp NoBlending mask)
  vertexShader v    = VertexOut v 1.0 () (Smooth v)
  vertexStream      = Fetch "stream4" Triangles (IV4F "position4")
  primitiveStream   = Transform vertexShader vertexStream
  fragmentStream    = Rasterize rasterCtx primitiveStream
  fragmentShader v  = let res = case choice of
                            A -> V4F 0.0 1.0 0.0 1.0
                            B -> V4F 1.0 0.0 0.0 1.0
                      in FragmentOut res
  frame             = Accumulate fragmentCtx PassAll fragmentShader fragmentStream fb
 in frame

emptyFB = FrameBuffer (colorImage1 (V4F 0.0 0.0 1.0 1.0))

main = ScreenOut (foldl' render emptyFB [A,B])
