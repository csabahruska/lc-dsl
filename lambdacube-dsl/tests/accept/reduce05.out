Pipeline {backend = OpenGL33, textures = [], samplers = [], targets = [RenderTarget {renderTargets = [(Color,Just (Framebuffer Color))]}], programs = [Program {programUniforms = fromList [], programStreams = fromList [("v",("position4",V4F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nin vec4 v ;\nsmooth out vec4 v0 ;\nvoid main() {\nv0 = v;\ngl_Position = v;\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nsmooth in vec4 v0 ;\nout vec4 f0 ;\nvoid main() {\nf0 = vec4 ( 1.0,0.0,0.0,1.0 );\n}\n"},Program {programUniforms = fromList [], programStreams = fromList [("v",("position4",V4F))], programInTextures = fromList [], programOutput = [("f0",V4F)], vertexShader = "#version 330 core\nin vec4 v ;\nsmooth out vec4 v0 ;\nvoid main() {\nv0 = v;\ngl_Position = v;\ngl_PointSize = 1.0;\n}\n", geometryShader = Nothing, fragmentShader = "#version 330 core\nsmooth in vec4 v0 ;\nout vec4 f0 ;\nvoid main() {\nf0 = vec4 ( 0.0,1.0,0.0,1.0 );\n}\n"}], slots = [Slot {slotName = "stream4", slotUniforms = fromList [], slotStreams = fromList [("position4",V4F)], slotPrimitive = Triangles, slotPrograms = [0,1]}], commands = [SetRenderTarget 0,ClearRenderTarget [(Color,VV4F (V4 0.0 0.0 1.0 1.0))],SetRasterContext (TriangleCtx CullNone PolygonFill NoOffset LastVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [ColorOp NoBlending (VV4B (V4 False True False False))]}),SetProgram 1,RenderSlot 0,SetRasterContext (TriangleCtx CullNone PolygonFill NoOffset LastVertex),SetAccumulationContext (AccumulationContext {accViewportName = Nothing, accOperations = [ColorOp NoBlending (VV4B (V4 True False False False))]}),SetProgram 0,RenderSlot 0]}